src/Core/src/Platform/iOS/MauiTextView.cs

```diff
@@ -172,6 +172,17 @@ void ShouldCenterVertically()
 				Maui.TextAlignment.End => new CGPoint(0, -Math.Max(1, availableSpace)),
 				_ => ContentOffset,
 			};
+
+			// Scroll the content to the cursor position if it is hidden by the keyboard
+			if (KeyboardAutoManagerScroll.IsKeyboardShowing)
+		    {
+		       var cursorRect = KeyboardAutoManagerScroll.FindCursorPosition();
+		       if (cursorRect.HasValue && cursorRect.Value.Bottom > KeyboardAutoManagerScroll.KeyboardFrame.Top)
+		       {
+		         var offset = cursorRect.Value.Bottom - KeyboardAutoManagerScroll.KeyboardFrame.Top;
+		         ContentOffset = new CGPoint(ContentOffset.X, ContentOffset.Y + offset);
+		       }
+		     }
```

Comment:
I think this is a great start on this one but I didn't like the bouncing that this issue created and still happens with this fix. I noticed that the ContentOffset is reset to zero every time we enter the LayoutSubviews call and this may contribute to the bouncing as well as having some interaction with the ContentInset changes from the ApplyContentInset method. I saw that there is also a `TextContainerInset` property that may be better to use for centering the text and from my testing, it looks like this behaves how we'd like it to and avoids any bouncing. Would you be able to try this out and see if it looks good to you? Perhaps some UITests would be good for this too!

```cs
		void ShouldCenterVertically()
		{
			var contentHeight = ContentSize.Height - TextContainerInset.Top - TextContainerInset.Bottom;
			var availableSpace = Bounds.Height - contentHeight * ZoomScale;
			if (availableSpace <= 0)
				return;

			TextContainerInset = VerticalTextAlignment switch
			{
				Maui.TextAlignment.Center => new UIEdgeInsets((nfloat)Math.Max(1, availableSpace / 2), TextContainerInset.Left, TextContainerInset.Bottom, TextContainerInset.Right),
				Maui.TextAlignment.End => new UIEdgeInsets((nfloat)Math.Max(1, availableSpace), TextContainerInset.Left, TextContainerInset.Bottom, TextContainerInset.Right),
				_ => TextContainerInset,
			};
		}
```

Reply:
@tj-devel709 , The given approach works well. I have tested this implementation using the following methods:

### With ScrollView
1. VerticalTextAlignment - Start
2. VerticalTextAlignment - Center
3. VerticalTextAlignment - End

### Without ScrollView
1. VerticalTextAlignment - Start
2. VerticalTextAlignment - Center
3. VerticalTextAlignment - End

I have included UI test for this Scenario.